Fixed: Issue 90

jvivian opened the pull request:
* Removed Required=true from @Arguments
* Supplied control statement in run method to handle -list_comparisons exception

AmplabJenkins commented:
Can one of the admins verify this patch?

jvivian commented on adam-cli/src/main/scala/org/bdgenomics/adam/cli/CompareADAM.scala:
```diff
@@ -173,13 +173,19 @@ class CompareADAM(protected val args: CompareADAMArgs) extends ADAMSparkCommand[
           println("\t%10s : %s".format(generator.name, generator.description))
       }
-
       return
```
Commenting this return statement out produces an error. What exactly does it do?

hannes-ucsc replied:
It terminate the run() method. Unless the caller has something else to do (which I don't think is the case) it will return also, and so on until the current thread's main run method returns. That will terminate the thread and if it is the last non-daemon thread, the equivalent of sys.exit(0) will be performed, i.e. the JVM process terminates with a zero status code.

hannes-ucsc commented on adam-cli/src/main/scala/org/bdgenomics/adam/cli/CompareADAM.scala:
```diff
@@ -173,13 +173,19 @@ class CompareADAM(protected val args: CompareADAMArgs) extends ADAMSparkCommand[
+      val parser = new CmdLineParser(options)
+      println("\n2 Arguments (files) must be given as input.\n")
+      parser.printUsage(System.err)
+      sys.exit(1)
```
You might want to compare this to how Args4j handles parsing errors. It might raise an exception instead of calling System.exit(). If that is the case, that behavior should be emulated.

fnothaft commented:
Jenkins, add to whitelist and test this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/565/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/566/
Test PASSed.

fnothaft commented:
@jvivian can you rebase this on master? I would merge but am getting merge conflicts.

fnothaft commented:
Also, can you squash these commits down to a single commit?

jvivian commented:
Should I close this request and open another? I've committed the changes to the master branch (as one commit).

fnothaft commented:
You should be able to do the squash and rebase on the original branch that the PR is from. If you do that and then push that branch, then you won't need to open a new PR.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/584/<h2>Build result: FAILURE</span></h2>GitHub pull request #560 of commit c0093adaceb7f0bab6a954b4e6c11c636f23c02e.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse c0093adaceb7f0bab6a954b4e6c11c636f23c02e^{commit} # timeout=10Checking out Revision c0093adaceb7f0bab6a954b4e6c11c636f23c02e (detached) > git config core.sparsecheckout # timeout=10 > git checkout -f c0093adaceb7f0bab6a954b4e6c11c636f23c02e > git rev-list d8e02d9f8f96c10595d575b7d0c5729f61cd8edc # timeout=10Triggering ADAM-prb ? 2.2.0,centosTriggering ADAM-prb ? 2.3.0,centosTriggering ADAM-prb ? 1.0.4,centosADAM-prb ? 2.2.0,centos completed with result SUCCESSADAM-prb ? 2.3.0,centos completed with result SUCCESSADAM-prb ? 1.0.4,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

fnothaft commented:
Jenkins, retest this please.

jvivian commented:
Sorry Frank, I'm in the meeting Matt is in -- I'll squash and rebase ASAP.

fnothaft commented:
@jvivian no worries! Take your time.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/585/<h2>Build result: ABORTED</span></h2>GitHub pull request #560 of commit c0093adaceb7f0bab6a954b4e6c11c636f23c02e.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse c0093adaceb7f0bab6a954b4e6c11c636f23c02e^{commit} # timeout=10Checking out Revision c0093adaceb7f0bab6a954b4e6c11c636f23c02e (detached) > git config core.sparsecheckout # timeout=10 > git checkout -f c0093adaceb7f0bab6a954b4e6c11c636f23c02e > git rev-list c0093adaceb7f0bab6a954b4e6c11c636f23c02e # timeout=10Triggering ADAM-prb ? 2.2.0,centosTriggering ADAM-prb ? 2.3.0,centosTriggering ADAM-prb ? 1.0.4,centosADAM-prb ? 2.2.0,centos completed with result ABORTEDADAM-prb ? 2.3.0,centos completed with result SUCCESSADAM-prb ? 1.0.4,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/587/<h2>Build result: FAILURE</span></h2>GitHub pull request #560 of commit e4f3520ec0cf9b75ac53e5ee8998c9fa42c91bc9 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/560/merge^{commit} # timeout=10Checking out Revision 66bb29ebde9d07ea2189b83f48e5c3ef80db8624 (detached) > git config core.sparsecheckout # timeout=10 > git checkout -f 66bb29ebde9d07ea2189b83f48e5c3ef80db8624 > git rev-list c0093adaceb7f0bab6a954b4e6c11c636f23c02e # timeout=10Triggering ADAM-prb ? 2.2.0,centosTriggering ADAM-prb ? 2.3.0,centosTriggering ADAM-prb ? 1.0.4,centosADAM-prb ? 2.2.0,centos completed with result FAILUREADAM-prb ? 2.3.0,centos completed with result FAILUREADAM-prb ? 1.0.4,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

jvivian commented:
I rebased and squashed, but I had to force push.  Given that my previous commits were built successfully, I'm not sure the current build failure is due to my commits.

fnothaft commented:
@jvivian this isn't an issue on your side; there's a known bug in our CI system that causes the system to intermittently fail to correctly check out a fork after a force push. We just try rerunning the tests until the CI system checks out the fork and builds correctly.

fnothaft commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/588/
Test PASSed.

fnothaft commented:
Thanks @jvivian! I've manually committed this via https://github.com/bigdatagenomics/adam/commit/1956ac9dc990b4e551e4764fc833bcf50ee8e720.
